src/core/resources/chains/chainMetadata.ts

```diff
@@ -1,14 +1,28 @@
+import { JsonRpcProvider } from '@ethersproject/providers';
 import { useQuery } from '@tanstack/react-query';
 
+import { proxyRpcEndpoint } from '~/core/providers/proxy';
 import {
   QueryConfig,
   QueryFunctionArgs,
   QueryFunctionResult,
   createQueryKey,
 } from '~/core/react-query';
-import { getChainMetadataRPCUrl } from '~/core/utils/chains';
 import { isValidUrl } from '~/core/utils/connectedApps';
 
+export const getChainMetadataRPCUrl = async ({
+  rpcUrl,
+}: {
+  rpcUrl?: string;
+}) => {
+  if (rpcUrl && isValidUrl(rpcUrl)) {
+    const provider = new JsonRpcProvider(proxyRpcEndpoint(rpcUrl, 0));
+    const network = await provider.getNetwork();
+    return { chainId: network.chainId };
```

Comment:
Is it possible to use viem ?

Reply:
in which sense?

this method is used when the user puts an url in the custom network form so we don't have that rpc as a network in wagmi, so this is how we're just fetching the chainid